lib/galaxy/datatypes/tabular.py

```diff
@@ -1779,3 +1779,169 @@ def set_meta(self, dataset, overwrite=True, skip=None, max_data_lines=7, **kwd):
+        super().__init__(**kwd)
+        self.column_names = [
+            "matches",
+            "misMatches",
+            "repMatches",
+            "nCount",
+            "qNumInsert",
+            "qBaseInsert",
+            "tNumInsert",
+            "tBaseInsert",
+            "strand",
+            "qName",
+            "qSize",
+            "qStart",
+            "qEnd",
+            "tName",
+            "tSize",
+            "tStart",
+            "tEnd",
+            "blockCount",
+            "blockSizes",
+            "qStarts",
+            "tStarts",
+        ]
+
+    def display_peek(self, dataset):
+        """Returns formated html of peek"""
+        return self.make_html_table(dataset, column_names=self.column_names)
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
+        """
+        PSL lines represent alignments, and are typically generated
+        by BLAT. Each line consists of 21 required fields, and track
+        lines may optionally be used to provide more information.
+
+        Fields are tab-separated, and all 21 are required.
+        Although not part of the formal PSL specification, track lines
+        may be used to further configure sets of features.  Track lines
+        are placed at the beginning of the list of features they are
+        to affect.
+
+        Rules for sniffing as True::
+
+            - There must be 21 columns on each fields line
+            - matches, misMatches repMatches, nCount, qNumInsert,
+              qBaseInsert, tNumInsert, tBaseInsert, strand, qSize, qStart,
+              qEnd, tName, tSize, tStart, tEnd, blockCount, blockSizes,
+              qStarts, tStarts  must be correct
+            - We will only check that up to the first 10 alignments are
+              correctly formatted.
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( '1.psl' )
+        >>> Psl().sniff( fname )
+        True
+        >>> fname = get_test_fname( '2.psl' )
+        >>> Psl().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'interval.interval' )
+        >>> Psl().sniff( fname )
+        False
+        >>> fname = get_test_fname( '2.txt' )
+        >>> Psl().sniff( fname )
+        False
+        >>> fname = get_test_fname( 'test_tab2.tabular' )
+        >>> Psl().sniff( fname )
+        False
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.ref.taxonomy' )
+        >>> Psl().sniff( fname )
+        False
+        """
+        count = 0
+        for line in file_prefix.line_iterator():
+            line = line.strip()
+            if not line:
+                break
+            if line:
+                if line.startswith("browser") or line.startswith("track"):
+                    # Skip track lines.
+                    continue
+                items = line.split("\t")
+                if len(items) != 21:
+                    return False
+                # matches
```

Comment:
Can you pop out the items that are not supposed to be non-negative integers, validate those, then just check the rest with
```python
if any(int(item) < 0 for item in items):
    raise Exception("Out of range")
```
?

Reply:
All of the above should be fixed - thanks @mvdbeek